File: VehicleLicenseApplication/src/main/java/com/app/service/UserService.java
package com.app.service;

import java.util.List;

import com.app.model.User;

public interface UserService {
	public String userRegistration(User user);

	public String userLogin(User user);

	public String changePassword(User user);

	public String forgotPassword(User user);

	public char[] generateOtp(int len);

	public List<User> getAllUsers();
}

File: Vehicle License src/src/app/service/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from 'src/models/User';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  private userRegistrationUrl= 'http://localhost:8080/user/registration';
  private userLogin  = 'http://localhost:8080/user/login';
  private changePasswordUrl= 'http://localhost:8080/user/changePassword';
  private forgotPassword= 'http://localhost:8080/user/forgotPassword';
  private viewAllUsers = 'http://localhost:8080/user/viewall';
  
  constructor(private http: HttpClient) { }

  userRegistration(user:User):Observable<any>{
    return this.http.post(this.userRegistrationUrl,user)
  }

}
